common-custom-user-data-maven-extension/src/main/java/com/gradle/CustomBuildScanEnhancements.java

```diff
@@ -0,0 +1,253 @@
+package com.gradle;
+
+import com.gradle.maven.extension.api.scan.BuildScanApi;
+
+import java.util.Properties;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.gradle.Utils.*;
+
+/**
+ * Adds a standard set of useful tags, links and custom values to all build scans published.
+ */
+final class CustomBuildScanEnhancements {
```

Comment:
One thing we could do is put this file in a shared library so that the maven extension and the gradle plugin can access it and we don't have to duplicate it like the spring project does: https://github.com/spring-io/gradle-enterprise-conventions

I'm guess we might not though because that would make it too complicated for customers to use the custom maven extension?

Reply:
Yexp, that's the reason we live with the duplication.

Reply:
We would also have to adapt to `com.gradle.maven.extension.api.scan.BuildScanApi` vs `com.gradle.scan.plugin.BuildScanExtension`, which would add complexity that isn't desired in a example app.